static/js/components/AutoComplete.js

```diff
@@ -407,84 +420,43 @@ class AutoComplete extends Component {
     const {prepareStyles} = this.context.muiTheme;
     const styles = getStyles(this.props, this.context, this.state);
 
-    const requestsList = [];
-    let filter = this.props.filter;
-    if (this.state.skipFilter) {
-      filter = AutoComplete.noFilter;
+    let requestsList = this.getFilteredDataSource();
+    if (maxSearchResults > 0) {
+      requestsList = requestsList.slice(0, maxSearchResults);
     }
 
-    dataSource.every((item, index) => {
-      switch (typeof item) {
-        case 'string':
-          if (filter(searchText, item, item)) {
-            requestsList.push({
-              text: item,
-              value: (
-                <MenuItem
-                  innerDivStyle={styles.innerDiv}
-                  value={item}
-                  primaryText={item}
-                  disableFocusRipple={disableFocusRipple}
-                  key={index}
-                />),
-            });
-          }
-          break;
-
-        case 'object':
-          if (item && typeof item.text === 'string') {
-            if (filter(searchText, item.text, item)) {
-              if (item.value.type && (item.value.type.muiName === MenuItem.muiName ||
-                 item.value.type.muiName === Divider.muiName)) {
-                requestsList.push({
-                  text: item.text,
-                  value: React.cloneElement(item.value, {
-                    key: index,
-                    disableFocusRipple: this.props.disableFocusRipple,
-                  }),
-                });
-              } else {
-                requestsList.push({
-                  text: item.text,
-                  value: (
-                    <MenuItem
-                      innerDivStyle={styles.innerDiv}
-                      primaryText={item.value}
-                      disableFocusRipple={disableFocusRipple}
-                      key={index}
-                    />),
-                });
-              }
-            }
-          }
-          break;
-
-        default:
-          // Do nothing
-      }
-
-      return !(maxSearchResults && maxSearchResults > 0 && requestsList.length === maxSearchResults);
-    });
+    let renderItem = option => (
+      <MenuItem
+        key={option.value}
+        primaryText={option.label}
+        value={option.value}
+        disableFocusRipple={disableFocusRipple}
+        innerDivStyle={styles.innerDiv}
+      />
+    );
```

Comment:
is it necessary/beneficial to define this inside of the `render` method? i think this would be cleaner if it was defined on the class and accepted a `divStyle` argument, et al. we would also avoid redefining this function every time `render` was called